### ISSUE TITLE
Looking for a more detailed explanation of the `isolation` property

### ISSUE THREAD

**Issue by RazrFalcon** [w3c/svgwg#871]
https://drafts.fxtf.org/compositing-1/#isolation

I'm trying to implement this property in [my library](https://github.com/RazrFalcon/resvg), but I'm having troubles understanding what `isolation` actually suppose to do. The spec only skims over it without much explanation or examples.
Based on my experiments, it seems like this property simply forces a new "layer"/_stacking context_ creation during rendering. Is it correct?

For example, a dummy group will be ignored by renderer. So the SVG below will be rendered just as `<rect/>`.

```xml
<g>
    <rect/>
</g>
```

But in this case:

```xml
<g opacity="0.5">
    <rect/>
</g>
```

a group cannot be skipped and will lead to a new "layer" creation.

Do I understand correctly, that `isolation:isolate` would prevent a rendered to skip a group in the first example?

**Comment by BigBadaboom** [comment]
I am not an expert, but I have implemented this myself.

Certain conditions automatically force isolation (a new layer, as you call it).  For example if the group has opacity, or a blend mode.  `isolation: isolate` forces a new layer in cases where it wouldn't automatically happen.

> Do I understand correctly, that isolation:isolate would prevent a rendered to skip a group in the first example?

Not necessarily. For simplicity, you may choose to implement it that way.  But in the case of a single `<rect>` in a dummy group, you shouldn't need to create a new layer.

**Comment by RazrFalcon** [comment]
So `isolation` is away to force, well, isolation for elements without using `opacity`, `filter`, `mask` and/or `clip-path`?

**Comment by BigBadaboom** [comment]
Yes. Sometimes you need to force isolation to get the desired result.  See the [example on the MDN page](https://developer.mozilla.org/en-US/docs/Web/CSS/isolation).
